x-pack/plugins/monitoring/init.js

```diff
@@ -10,46 +10,36 @@ import { initMonitoringXpackInfo } from './server/init_monitoring_xpack_info';
 import { initKibanaMonitoring } from './server/kibana_monitoring';
 
 /**
- * Initialize the Kibana Monitoring plugin by starting up asynchronous server
- * tasks, based on user-defined configuration
- * - webserver route handling
- * - monitoring cluster health checker
- * - instantiation of an elasticsearch-js client exposed as a server plugin object
- * - start kibana ops monitoring loop
- * - start monitoring cluster x-pack license and features check loop
+ * Initialize the Kibana Monitoring plugin by starting up asynchronous server tasks
+ * - [1] instantiation of an elasticsearch-js client exposed as a server plugin object
+ * - [2] start monitoring cluster x-pack license and features check
+ * - [3] webserver route handling
+ * - [4] start the internal monitoring collectors
+ * - [5] expose the monitoring collector object for other plugins to register with
+ * - [6] set monitoring plugin status to green
  * @param monitoringPlugin {Object} Monitoring UI plugin
  * @param server {Object} HapiJS server instance
  */
 export const init = (monitoringPlugin, server) => {
+  monitoringPlugin.status.yellow('Initializing');
   const xpackMainPlugin = server.plugins.xpack_main;
 
   xpackMainPlugin.status.once('green', async () => {
     const config = server.config();
     const uiEnabled = config.get('xpack.monitoring.ui.enabled');
-    const features = [];
-    const onceMonitoringGreen = callbackFn => monitoringPlugin.status.once('green', () => callbackFn()); // avoid race condition in things that require ES client
 
     if (uiEnabled) {
-      // Instantiate the dedicated ES client
-      features.push(instantiateClient(server));
-
-      // route handlers depend on xpackInfo (exposed as server.plugins.monitoring.info)
-      onceMonitoringGreen(async () => {
-        await initMonitoringXpackInfo(server);
-      });
-
-      // Require only routes needed for ui app
-      features.push(requireUIRoutes(server));
+      await instantiateClient(server); // Instantiate the dedicated ES client
+      await initMonitoringXpackInfo(server); // Route handlers depend on this for xpackInfo
+      await requireUIRoutes(server);
     }
 
-    // Send Kibana usage / server ops to the monitoring bulk api
     if (config.get('xpack.monitoring.kibana.collection.enabled')) {
```

Comment:
I've started this work in the 2nd PR to come after this one: https://github.com/elastic/kibana/pull/18894 It might be awhile before that gets filed, but I'll be sure to test what happens when Monitoring is disabled but Reporting is enabled.